pkg/allocator/localkeys.go

```diff
@@ -51,24 +51,26 @@ func newLocalKeys() *localKeys {
 // allocate creates an entry for key in localKeys if needed and increments the
 // refcnt. The value associated with the key must match the local cache or an
 // error is returned
-func (lk *localKeys) allocate(key string, val idpool.ID) (idpool.ID, error) {
+// Note: keyString is expected to be key.GetKey(). We allow it to be passed in
+// to avoid creating large strings repeatdly.
+func (lk *localKeys) allocate(keyString string, key AllocatorKey, val idpool.ID) (idpool.ID, error) {
```

Comment:
This specific function is almost exactly the same. The only difference is that the key was converted from a string to AllocatorKey. We also pass in `keyString` (I assume to avoid recreating the string repeatedly, as callers of this function use the key as a string a few times).
Did you mean that key is no longer a string? I could try to undo the abstraction but I'm not sure how much of it is removable this way. In any case, I've removed the extra parameter, so the only change is that now `key` is an `AllocatorKey`.

Reply:
My main concern is that there are small changes interspersed throughout code-shuffling, which means I may gloss over something which is small, but significant, in terms of behavior. Being able to know for sure that the only thing that has changed in a patch is the location of code rather than the code itself, is useful so that functionality changes can be analyzed more closely.

Reply:
Ok, @ianvernon I did manage an intermediate commit (https://github.com/cilium/cilium/pull/8496/commits/736725f69dd7eecdaf6575461249510008f8ee69) that should be only the code copies. It isn't quite correct, though. The tests are commented out completely because they caused import loops that I couldn't fix. Most of the differences should be renaming things to be in new packages, but it isn't all so straighforward. Does it help at all?